### PR TITLE
cpack: add apt package manager prerm script

### PR DESCRIPTION
**Pull request by ahspw**
### Description
When we install packages with apt, rpm or other package managers, some bash scripts with (prerm, preinst, postrm and postinst) extensions would be execute (if exists) that do some tasks like copy config files, create directories etc before or after unpacking package so the main program can run in the better environment.

### Issue
If remove and purge package without stop (and | or) disable it, OS still shows the service is running and can cause issues after rebooting.

### Added feature
In this branch, the prerm script is added to prevent those issues. (Built and tested on ubuntu:20.04)

### CODE REVIEW

File: cpack/debian/prerm [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e
+
+if [[ $1 == "remove" ]]; then
+    systemctl stop fluent-bit.service || true
+    systemctl disable fluent-bit.service || true
```
Comment (cosmo0920) [inline cpack/debian/prerm:7]: Shouldn't we check whether service or systemctl command is effective on the Debian box?

Comment (ahspw) [reply cpack/debian/prerm:7]: Sure! I will test it on Debian 11 and 12 and reply back with results.

Comment (ahspw) [reply cpack/debian/prerm:7]: Tests have been successfully done on debian bullseye and bookworm versions.
After Debian 8 systemd is the default init system for debian ([Debian Reference](https://wiki.debian.org/systemd#Introduction)) so systemctl would not make any issues.

File: cpack/debian/prerm [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (aminvakil) [inline cpack/debian/prerm:1]:
```suggestion
#!/bin/sh
```

**Review by aminvakil — COMMENTED** [review]
This should be investigated more whether this is the right choice to do or not, take fail2ban.prem for example:
```
#!/bin/sh
set -e
# Automatically added by dh_installinit/13.3.4
if [ -x "/etc/init.d/fail2ban" ] && [ "$1" = remove ]; then
        invoke-rc.d fail2ban stop || exit 1
fi
# End automatically added section

# Automatically added by dh_python3
if which py3clean >/dev/null 2>&1; then
        py3clean -p fail2ban 
else
        dpkg -L fail2ban | perl -ne 's,/([^/]*)\.py$,/__pycache__/\1.*, or next; unlink $_ or die $! foreach glob($_)'
        find /usr/lib/python3/dist-packages/ -type d -name __pycache__ -empty -print0 | xargs --null --no-run-if-empty rmdir
fi

# End automatically added section
```

First it uses `/bin/sh` instead of `/bin/bash` and it seems some stuff would be handled by `dh_installinit` which will be automatically added to `prerm` files of package.

So maybe it's better to rely on debscripts scripts to run these stuff instead of executing `systemctl disable` and `systemctl stop` manually.

**Comment by aminvakil** [comment]
Also `netfilter-persistent.prem` on Debian 10 Buster and Debian 11 Bullseye:

Debian 10 Buster:
```
#!/bin/sh
set -e
# Automatically added by dh_installinit/12.1.1
if [ -x "/etc/init.d/netfilter-persistent" ] && [ "$1" = remove ]; then
        invoke-rc.d netfilter-persistent stop || exit 1
fi
# End automatically added section
```

Debian 11 Bullseye:
```
#!/bin/sh
set -e
# Automatically added by dh_installsystemd/13.3.3
if [ -d /run/systemd/system ] && [ "$1" = remove ]; then
        deb-systemd-invoke stop 'netfilter-persistent.service' >/dev/null || true
fi
# End automatically added section
# Automatically added by dh_installinit/13.3.3
if [ -x "/etc/init.d/netfilter-persistent" ] && [ "$1" = remove ]; then
        invoke-rc.d --skip-systemd-native netfilter-persistent stop || exit 1
fi
# End automatically added section
```

This is more suggesting that these stuff should be handled using debhelper itself.

File: cpack/debian/prerm (hunk 1 above)
Comment (cosmo0920) [reply cpack/debian/prerm:7]: Sounds great. So, we shouldn't consider the outdated Debian system. Let's leave as-is.

**Pull request closed without merging**